src/UserAction/StepExecutorTrait.php

```diff
@@ -102,18 +105,53 @@ protected function setFormField(Form $form, array $fields, string $step): Form
         foreach ($fields as $k => $val) {
             $form->getControl($k)->set($val);
         }
+
         $this->hook(self::HOOK_STEP, [$step, $form]);
 
         return $form;
     }
 
+    /**
+     * Set model for userAction arguments.
+     * Override existing argument with model definition.
+     */
+    protected function getModelForArgs(): Model
+    {
+        $args = $this->getAction()->args;
+        if (array_key_exists('__atk_model', $args)) {
+            /** @var Model $argsModel */
+            $argsModel = Factory::factory($args['__atk_model']);
+            // if seed is supplied, we need to initialize.
+            if (!$argsModel->_initialized) {
+                $argsModel->invokeInit();
+            }
+
+            unset($args['__atk_model']);
+        } else {
+            $argsModel = new Model(new Array_([]));
+        }
+
+        foreach ($args as $key => $val) {
+            $argsModel->addField($key, $val);
+        }
+
+        // set userAction args using model field.
+        foreach ($argsModel->getFields('editable') as $k => $field) {
```

Comment:
```suggestion
        $this->cloneArgs = [];
        foreach ($argsModel->getFields('editable') as $k => $field) {
```
and no default value in the property declaration

Reply:
see https://github.com/atk4/ui/pull/1687/files#diff-bbd75fe6645c71d2c9892c71761a2044e9aa673eeb41f5f5f4a2abd26ba677d1R76

property $cloneArgs is already define as an empty array

Reply:
but is it correct, that if this method is called, `[]` default value is valid?

if not, `null` should be the default property value

Reply:
[] is needed as the cloneArg are collected to execute the action.
see: https://github.com/atk4/ui/pull/1687/files#diff-bbd75fe6645c71d2c9892c71761a2044e9aa673eeb41f5f5f4a2abd26ba677d1R527

Reply:
that is exactly my point :)

is `[]` valid there? it seems UNINITIALIZED to me, thus not valid

Reply:
Why do you say it is not initialized? It is when property is declare.

Reply:
I mean from app perspective, not language perspective. From brief look, it seems to be args are either set using `getModelForArgs` method or UNINITIALIZED - and the property value must be `null` not `[]`.

Reply:
not necessary - if UserAction does not contains arguments - like an Edit action for example - the $cloneArgs is not initialized.
But call to get $cloneArgs will always be call to execute the action. This is what happen if I set it to null

<img width="879" alt="Screen Shot 2021-11-05 at 10 22 03 AM" src="https://user-images.githubusercontent.com/2204478/140525952-0b88da98-78d1-4c2b-a6fc-44679f4f3167.png">

Therefore using an empty array as default value is needed.

Reply:
refactor method name and cloneArg  initialisation

Reply:
> not necessary - if UserAction does not contains arguments

thanks, with `$this->cloneArgs = [];` locally is right now :)